kubetest/kops.go

```diff
@@ -179,12 +215,38 @@ func newKops(provider, gcpProject, cluster string) (*kops, error) {
 	if err := os.Setenv("AWS_SSH_KEY", sshKey); err != nil {
 		return nil, err
 	}
-	// ZONE is required by the AWS e2e tests.
-	zones := strings.Split(*kopsZones, ",")
+
+	// zones are required by the kops e2e tests.
+	var zones []string
+
+	// if zones is set to zero and gcp project is not set then pick random aws zone
+	if *kopsZones == "" && provider == "aws" {
+		// Initialize global pseudo random generator.
+		// We do not need to intialize it if kopsZones is passed in or we are running the other tests.
+		// That is why I did not put it in main func.
```

Comment:
I think it should go into the main func, it's something that should be done once in every program and then we forget about it